tests/lax_scipy_special_functions_test.py

```diff
@@ -265,6 +268,18 @@ def testBetaParameterDeprecation(self):
       with self.assertRaises(TypeError), self.assertWarns(DeprecationWarning):
         lsp_special.beta(b=1, y=1)
 
+  def testFresnelSingleAndDoublePrecision(self):
```

Comment:
What does this test cover that the `op_record` test above does not?

Reply:
I did that to check that single and double precision would both work, as you questioned, but I did not know wether the op_record would already test that.

Reply:
`op_record` already does everything that is in this test, I believe.

My comment wasn't really about test coverage, but rather a theoretical/algorithmic question. Do we expect this expansion to work in float32? Can you think of corner cases that are untested that might have issues?

Reply:
Indeed, it does. I will have the test removed by tomorrow then :)

For the cornecases, I don’t see any obvious one. But if you have an example from another function, maybe that could give us some ideas?

Reply:
I just removed the redundant test.

Reply:
Well, for example #17038 is currently blocked because the bessel function approximations don't converge in float32. This implementation looks quite similar to that, which is why I brought it up.

Reply:
I see no particular reasons that it might fail, and no documented use of a single precision version. If you prefer, I can also implement the [single-precision `fresnelf`](https://github.com/jeremybarnes/cephes/blob/master/single/fresnlf.c) from Cephes, if you think that is safer. Reading the implementation, it differs quite a bit from the double precision one.

I tried to understand how SciPy deals with `float32`, but it seems that they would just cast the data type to `float64`, but it isn't easy to be sure about what they actually do (at least to what I have found in their codebase).

Reply:
scipy generally deals with float32 by casting the inputs to float64, computing the function in float64, and then casting the outputs back to float32. That's not an option in JAX.

Reply:
Then, should I implement a `float32`-specific version to be sure that it is made for single precision?

Reply:
Maybe - it might be that the float64 expansion works just fine when cast to float32. But we'd need to explore that a bit to be sure. e.g. choose a range of inputs, compute the result in both float64 and float32, and confirm that the float32 output is within a reasonable range of the expected value given float32 precision.

Reply:
But wasn’t what my previous test was doing? I computed using float32 with JAX, and compared it to the float32 version of SciPy with a relative error close to eps.

Reply:
Yeah, that's also what the `op_record` test infrastructure does. It's testing with `tol=1E-6`, which is about 8 ULP or so; maybe that's sufficient. 

But you also said that cephes float32 looks very different from float64. That suggests there's something deeper here: do you understand what that is?

Reply:
I see. My previous tests covered the `[-10, 10]` range, as mentioned in the original docstring (see [here](https://github.com/scipy/scipy/blob/76bf366486b0583dca789823878b80d610eb571b/scipy/special/xsf/cephes/ellik.h#L41-L47)). I have looked at other similar libraries, like TensorFlow, and they only have [one `fresnel_cos` and `fresnel_sin` implementation](https://github.com/tensorflow/tensorflow/blob/52b41bd227b6f280942f04d92076116e62b168e5/tensorflow/core/kernels/special_math/special_math_op_misc_impl.h#L410-L596): the same as the one propose here. If I have understood the code well enough, `Scalar` can either be  `float32` or `float64`, and *I doubt that TensorFlow would automatically promote `float32` to `float64`, compute the Fresnel integrals, and cast the dtype back to `float32`*, as this would be a major memory increase (usually a bottleneck in ML).

Last, I found [that paper](https://link.springer.com/content/pdf/10.1007/s11075-023-01654-2.pdf) (*Calculation of Fresnel integrals of real and complex
arguments up to 28 signifcant digits*, by **Mofreh R. Zaghloul and Leen Alrawas**) that shows the `f` and `g` (eq. 21 and 22) functions that are expanded into series (but the series used are different here, because they aim at a much higher precision).

The difference in power series chosen by Cephes for `float32` and `float64`, that I cannot explain. Except for the smaller amount of computations performed with the `float32` version, that could be justified for the need of less accuracy than the `float64` version.

Reply:
See my other coment: I think adding test coverage for large values will probably be sufficient.